src/gc-pages.c

```diff
@@ -107,6 +108,7 @@ NOINLINE jl_gc_pagemeta_t *jl_gc_alloc_page(void) JL_NOTSAFEPOINT
     // try to get page from `pool_freed`
     meta = pop_lf_page_metadata_back(&global_page_pool_freed);
     if (meta != NULL) {
+        jl_atomic_fetch_add_relaxed(&gc_heap_stats.bytes_mapped, GC_PAGE_SZ);
```

Comment:
This seems to be already incremented by jl_gc_try_alloc_pages_, and not decremented when we added them to pool_freed, resulting in the bytes_mapped counting becoming a significant overestimate of the actual usable pages?

Reply:
Right, this is something that bit me. The new allocator tries to revive madvised pages from the `pool_freed`. You can see that in `gc_sweep_page` we push them. I was getting a weird thing where I saw that pages were being freed but not reallocated and it was the reviving part here.